### PR TITLE
[vcpkg baseline][llvm] Fix file conflicts for libiomp5md.dll

### PR DESCRIPTION
**Pull request by FrankXie05**
Fixes vcpkg pipeline issue `llvm` conflict with `intel-mkl`:

```
error: The following files are already installed in D:/installed/x64-windows and are in conflict with llvm:x64-windows
Installed by intel-mkl:x64-windows    
bin/libiomp5md.dll
    debug/bin/libiomp5md.dll
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
This will generate valid binary artifacts at build time. But it won't resolve conflicts when installing the artifacts from the cache.
And even worse, replacing the installed DLL looks like asking for DLL hell because they probably constitute different runtimes.

**Comment by Neumann-A** [comment]
Patch the output name to have a suffix like `llvm-`

File: ports/llvm/0012-fix-file-conficts.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
++++ b/openmp/runtime/src/CMakeLists.txt
+@@ -206,7 +206,7 @@ if(NOT WIN32)
+     COMMAND ${CMAKE_COMMAND} -E create_symlink ${LIBOMP_LIB_FILE}
+       libgomp${LIBOMP_LIBRARY_SUFFIX}
+     COMMAND ${CMAKE_COMMAND} -E create_symlink ${LIBOMP_LIB_FILE}
+-      libiomp5${LIBOMP_LIBRARY_SUFFIX}
++      llvm-libiomp5${LIBOMP_LIBRARY_SUFFIX}
+     WORKING_DIRECTORY ${LIBOMP_LIBRARY_DIR}
+   )
+ endif()
```
Comment (Neumann-A) [inline ports/llvm/0012-fix-file-conficts.patch:26]: Really? Just deactivate the symlink here.

File: ports/llvm/0012-fix-file-conficts.patch [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@@ -355,7 +355,7 @@ if(WIN32)
+   install(TARGETS omp RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
+   install(TARGETS ${LIBOMP_IMP_LIB_TARGET} ARCHIVE DESTINATION "${OPENMP_INSTALL_LIBDIR}")
+   # Create aliases (regular copies) of the library for backwards compatibility
+-  set(LIBOMP_ALIASES "libiomp5md")
++  set(LIBOMP_ALIASES "llvm-libiomp5md")
+   foreach(alias IN LISTS LIBOMP_ALIASES)
+     install(CODE "execute_process(COMMAND \"\${CMAKE_COMMAND}\" -E copy \"${LIBOMP_LIB_FILE}\"
+       \"${alias}${LIBOMP_LIBRARY_SUFFIX}\" WORKING_DIRECTORY \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}\")")
```
Comment (Neumann-A) [inline ports/llvm/0012-fix-file-conficts.patch:35]: this also seems like something which could be deactivated

File: ports/llvm/0012-fix-file-conficts.patch [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+@@ -368,7 +368,7 @@ else()
+ 
+   if(${LIBOMP_INSTALL_ALIASES})
```
Comment (Neumann-A) [inline ports/llvm/0012-fix-file-conficts.patch:38]: this looks like `-DLIBOMP_INSTALL_ALIASES=OFF` would be enough ?

Comment (FrankXie05) [reply ports/llvm/0012-fix-file-conficts.patch:38]: Do we want to disable the entire `LIBOMP`? Because it's just a name conflict for `libiomp5md`? :)

**Comment by FrankXie05** [comment]
Modifying the patch.

File: ports/llvm/portfile.cmake [hunk 4]
```diff
@@ -14,6 +14,7 @@ vcpkg_from_github(
         0009-fix-tools-install-path.patch
         0010-fix-libffi.patch
         0011-fix-install-bolt.patch
+        0012-fix-disable-libomp.patch
```
Comment (Neumann-A) [inline ports/llvm/portfile.cmake:17]:
```suggestion
        0012-disable-libomp-aliases.patch
```

**Pull request merged**